src/portal-api.ts

```diff
@@ -362,3 +380,31 @@ export const fetchPortalData = async (): Promise<IPortalData> => {
   };
   return rawPortalData;
 };
+
+// metadata for saving and loading work for anonymous users, not actually loaded from the portal
+export const anonymousPortalData = () => {
+  const runKey = queryValue("runKey") || uuidv4();
+  setQueryValue("runKey", runKey);
```

Comment:
Given that `setQueryValue()` can potentially have side effects, I'd be inclined to avoid it when not necessary, e.g. something like:
```
  let runKey = queryValue("runKey");
  if (!runKey) {
    runKey = uuidv4();
    setQueryValue("runKey", runKey);
  }
```

Reply:
Makes sense, you're right. (Although if we don't trust `setQueryValue` to not have problematic have side-effects, we ought to worry about those even when we're adding the runKey.)